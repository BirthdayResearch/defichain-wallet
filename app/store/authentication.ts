import { createSlice, PayloadAction } from '@reduxjs/toolkit'

export interface Authentication<T> {
  consume: (passphrase: string) => Promise<T> // also serve as passphrase validation logic
  onAuthenticated: (result: T) => Promise<void>
  // OPTIONAL error handler
  // (CRITICAL invalid passphare error is auto handled, include wipe wallet if necessary)
  onError?: (e: Error) => void

  // messages in passcode UI
  // mandatory, default message is for signing transaction
  message: string
  loading: string
}

const initialState: { authentication?: Authentication<any> } = {}

export const authentication = createSlice({
  name: 'authentication',
  initialState,
  reducers: {
    prompt: (state, action: PayloadAction<Authentication<any>>) => {
      state.authentication = action.payload
    },
    dismiss: (state) => {
      delete state.authentication
    }
  }
})
