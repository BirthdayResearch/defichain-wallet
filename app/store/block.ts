import { createSlice, PayloadAction } from '@reduxjs/toolkit'

interface BlockState {
  count?: number
}

const initialState: BlockState = {
  count: undefined
}

export const block = createSlice({
  name: 'block',
  initialState,
  reducers: {
    updateBlock: (state, action: PayloadAction<{ count: number }>) => {
      state.count = action.payload.count
    }
  }
})
