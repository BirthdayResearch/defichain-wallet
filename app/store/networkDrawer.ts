import { CTransactionSegWit } from '@defichain/jellyfish-transaction'
import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit'

export interface Transaction {
  broadcasted: boolean
  signed: CTransactionSegWit
  title?: string
}

export interface NetworkDrawerState {
  transactions: Transaction[]
  height: number
  err?: Error
}

const initialState: NetworkDrawerState = {
  transactions: [],
  height: 49,
  err: undefined
}

export const networkDrawer = createSlice({
  name: 'networkDrawer',
  initialState,
  reducers: {
    setHeight: (state, action: PayloadAction<number>) => {
      state.height = action.payload
    },
    queueTransaction: (state, action: PayloadAction<Transaction>) => {
      state.transactions = [...state.transactions, action.payload]
    },
    closeNetworkDrawer: (state) => {
      state.transactions = []
      state.err = undefined
    },
    setError: (state, action: PayloadAction<Error>) => {
      state.err = action.payload
    },
    popTransaction: (state) => {
      if (state.transactions.length > 1) {
        state.transactions.shift()
        state.transactions = [...state.transactions]
      }
    }
  }
})

export const isDrawerOpenSelector = createSelector([(state: NetworkDrawerState) => state.transactions, (state: NetworkDrawerState) => state.err],
  (transactions, err) => transactions?.length > 0 || err !== undefined)

export const firstTransactionSelector = createSelector((state: NetworkDrawerState) => state.transactions, (transactions) => transactions[0])
