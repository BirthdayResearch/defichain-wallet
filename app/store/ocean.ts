import { CTransactionSegWit } from '@defichain/jellyfish-transaction'
import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit'

export interface OceanTransaction {
  broadcasted: boolean
  signed: CTransactionSegWit
  title?: string
}

export interface OceanState {
  transactions: OceanTransaction[]
  height: number
  err?: Error
}

const initialState: OceanState = {
  transactions: [],
  height: 49,
  err: undefined
}

export const ocean = createSlice({
  name: 'ocean',
  initialState,
  reducers: {
    setHeight: (state, action: PayloadAction<number>) => {
      state.height = action.payload
    },
    queueTransaction: (state, action: PayloadAction<OceanTransaction>) => {
      state.transactions = [...state.transactions, action.payload]
    },
    closeOceanInterface: (state) => {
      state.transactions = []
      state.err = undefined
    },
    setError: (state, action: PayloadAction<Error>) => {
      state.err = action.payload
    },
    popTransaction: (state) => {
      if (state.transactions.length > 1) {
        state.transactions.shift()
        state.transactions = [...state.transactions]
      }
    }
  }
})

export const isDrawerOpenSelector = createSelector([(state: OceanState) => state.transactions, (state: OceanState) => state.err],
  (transactions, err) => transactions?.length > 0 || err !== undefined)

export const firstTransactionSelector = createSelector((state: OceanState) => state.transactions, (transactions) => transactions[0])
