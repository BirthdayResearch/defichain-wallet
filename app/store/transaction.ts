import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { VMTransaction } from '../screens/AppNavigator/screens/Transactions/screens/stateProcessor'

type LoadingState = 'idle' | 'loading' | 'loadingMore' | 'success' | 'background' | 'error'

export interface TransactionState {
  transactions: VMTransaction[]
  loadingState: LoadingState
  error: string | null
  loadMoreToken: string | undefined
}

const initialState: TransactionState = {
  transactions: [],
  loadingState: 'idle',
  error: null,
  loadMoreToken: undefined
}

export const transaction = createSlice({
  name: 'transaction',
  initialState,
  reducers: {
    setTransactions: (state, action: PayloadAction<VMTransaction[]>) => {
      state.transactions = action.payload
      state.loadingState = 'success'
    },
    addTransactions: (state, action: PayloadAction<VMTransaction[]>) => {
      state.transactions = state.transactions.concat(action.payload)
      state.loadingState = 'success'
    },
    setLoadingState: (state, action: PayloadAction<LoadingState>) => {
      state.loadingState = action.payload
    },
    setLoadMoreToken: (state, action: PayloadAction<string | undefined>) => {
      state.loadMoreToken = action.payload
    },
    setError: (state, action: PayloadAction<any>) => {
      state.error = action.payload
    }
  }
})
