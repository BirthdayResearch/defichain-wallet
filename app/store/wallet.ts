import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpairs'
import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit'
import BigNumber from 'bignumber.js'

export interface WalletToken extends AddressToken {
  displaySymbol: string
  avatarSymbol: string
}

export interface WalletState {
  utxoBalance: string
  tokens: WalletToken[]
  poolpairs: DexItem[]
}

export interface DexItem {
  type: 'your' | 'available'
  data: PoolPairData
}

const initialState: WalletState = {
  utxoBalance: '0',
  tokens: [],
  poolpairs: []
}

const tokenDFI: WalletToken = {
  id: '0',
  symbol: 'DFI',
  symbolKey: 'DFI',
  isDAT: true,
  isLPS: false,
  amount: '0',
  name: 'DeFiChain',
  displaySymbol: 'DFI (Token)',
  avatarSymbol: 'DFI'
}

const utxoDFI: WalletToken = {
  ...tokenDFI,
  id: '0_utxo',
  displaySymbol: 'DFI (UTXO)',
  avatarSymbol: '_UTXO'
}

export const wallet = createSlice({
  name: 'wallet',
  initialState,
  reducers: {
    setTokens: (state, action: PayloadAction<AddressToken[]>) => {
      state.tokens = action.payload.map((t) => {
        let displaySymbol = t.symbol
        let avatarSymbol = t.symbol
        if (t.id === '0') {
          t.name = 'DeFiChain'
          displaySymbol = 'DFI (Token)'
        }
        if (t.id === '0_utxo') {
          displaySymbol = 'DFI (UTXO)'
          avatarSymbol = '_UTXO'
        }
        return { ...t, displaySymbol, avatarSymbol }
      })
    },
    setUtxoBalance: (state, action: PayloadAction<string>) => {
      state.utxoBalance = action.payload
    },
    setPoolPairs: (state, action: PayloadAction<DexItem[]>) => {
      state.poolpairs = action.payload
    }
  }
})

const rawTokensSelector = createSelector((state: WalletState) => state.tokens, (tokens) => {
  const rawTokens = []
  if (!tokens.some((t) => t.id === '0_utxo')) {
    rawTokens.push(utxoDFI)
  }
  if (!tokens.some((t) => t.id === '0')) {
    rawTokens.push(tokenDFI)
  }
  return [...rawTokens, ...tokens]
})

export const tokensSelector = createSelector([rawTokensSelector, (state: WalletState) => state.utxoBalance], (tokens, utxoBalance) => tokens.map((t) => {
  if (t.id === '0_utxo') {
    return { ...t, amount: new BigNumber(utxoBalance).toFixed(8) }
  }
  return t
}))
