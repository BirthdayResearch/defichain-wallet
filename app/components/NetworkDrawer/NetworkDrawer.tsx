import { MaterialIcons } from '@expo/vector-icons'
import React, { useRef } from 'react'
import { ActivityIndicator, Animated, Linking, TouchableOpacity, View } from 'react-native'
import { useDispatch, useSelector } from 'react-redux'
import tailwind from 'tailwind-rn'
import { Text } from '..'
import { PrimaryColor, PrimaryColorStyle } from '../../constants/Theme'
import { RootState } from '../../store'
import { networkDrawer } from '../../store/networkDrawer'
import { translate } from '../../translations'

async function handlePress (txid: string): Promise<void> {
  // TODO(thedoublejay) explorer URL
  const url = `https://explorer.defichain.io/#/DFI/mainnet/tx/${txid}`
  const supported = await Linking.canOpenURL(url)
  if (supported) {
    await Linking.openURL(url)
  }
}

/**
 * @description - Global component to be used for async calls, network errors etc. This component is positioned above the bottom tab.
 * Need to get the height of bottom tab via `useBottomTabBarHeight()` hook to be called on screen.
 * @example
 *          const height = useBottomTabBarHeight()
 *          // Accepts partial state
 *          dispatch(openNetworkDrawer({ isOpen: true, isLoading: true, height }))
 * */
export function NetworkDrawer (): JSX.Element {
  const { height, isLoading, txid, title, isOpen } = useSelector((state: RootState) => state.networkDrawer)
  const dispatch = useDispatch()

  const slideAnim = useRef(new Animated.Value(0)).current
  Animated.timing(slideAnim, { toValue: isOpen ? height : 0, duration: 300, useNativeDriver: false }).start()

  return (
    <Animated.View
      style={[tailwind('bg-white px-5 py-3 flex-row absolute w-full items-center border-t border-gray-200 z-10'), {
        bottom: slideAnim,
        height: 75
      }]}
    >
      {
        isLoading ? <ActivityIndicator color={PrimaryColor} />
          : <MaterialIcons name='check-circle' size={20} color='#02B31B' />
      }
      <View style={tailwind('flex-grow mr-1 justify-center items-center text-center')}>
        <Text style={tailwind('text-sm font-bold')}>{translate('screens/NetworkDrawer', title)}</Text>
        {
          txid !== undefined && (
            <TouchableOpacity
              testID='networkDrawer_explorer' style={tailwind('flex-row bg-white p-1 items-center')}
              onPress={async () => await handlePress(txid)}
            >
              <Text style={[PrimaryColorStyle.text, tailwind('text-sm font-medium mr-1')]}>
                {`${txid.substring(0, 15)}...`}
              </Text>
              <MaterialIcons name='open-in-new' size={18} color={PrimaryColor} />
            </TouchableOpacity>
          )
        }
      </View>
      {
        !isLoading && (
          <TouchableOpacity
            testID='networkDrawer_close' onPress={() => {
              dispatch(networkDrawer.actions.openNetworkDrawer({ isOpen: false }))
            }} style={tailwind('px-2 py-1 rounded border border-gray-300 rounded flex-row justify-center items-center')}
          >
            <Text style={[PrimaryColorStyle.text, tailwind('text-sm')]}>
              {translate('screens/NetworkDrawer', 'OK')}
            </Text>
          </TouchableOpacity>
        )
      }
    </Animated.View>
  )
}
