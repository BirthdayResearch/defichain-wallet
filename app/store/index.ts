/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { authentication } from './authentication'
import { block } from './block'
import { ocean } from './ocean'
import { transaction } from './transaction'
import { transactionQueue } from './transaction_queue'
import { wallet } from './wallet'

/**
 * RootState for DeFiChain Wallet App
 *
 * All state reducer in this store must be designed for global use and placed in this
 * directory as such. Reducer that are not meant to be global must not be part of
 * RootState.
 *
 * Non-global state should be managed independently within its own React Component.
 */
export function initializeStore () {
  return configureStore({
    reducer: {
      block: block.reducer,
      wallet: wallet.reducer,
      ocean: ocean.reducer,
      transactionQueue: transactionQueue.reducer,
      authentication: authentication.reducer,
      transaction: transaction.reducer
    },
    middleware: [
      ...getDefaultMiddleware({ serializableCheck: false })
    ]
  })
}

export type RootStore = ReturnType<typeof initializeStore>
export type RootState = ReturnType<RootStore['getState']>
