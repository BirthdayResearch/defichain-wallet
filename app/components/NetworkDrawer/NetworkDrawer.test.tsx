import { configureStore } from "@reduxjs/toolkit";
import { fireEvent, render } from "@testing-library/react-native";
import * as React from "react";
import { Linking } from "react-native";
import { Provider } from "react-redux";
import { RootState } from "../../store";
import { networkDrawer } from "../../store/networkDrawer";
import { NetworkDrawer } from "./NetworkDrawer";

describe('networkDrawer', () => {
  it('should match snapshot', async () => {
    const initialState: Partial<RootState> = {
      networkDrawer: {
        height: 49,
        isOpen: false,
        title: 'Loading...',
        isLoading: false,
        txid: ''
      }
    };
    const store = configureStore({
      preloadedState: initialState,
      reducer: { networkDrawer: networkDrawer.reducer }
    })
    const component = (
      <Provider store={store}>
        <NetworkDrawer />
      </Provider>
    );
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('should display loading state', async () => {
    const initialState: Partial<RootState> = {
      networkDrawer: {
        height: 49,
        isOpen: true,
        title: 'Loading...',
        isLoading: true,
        txid: ''
      }
    };
    const store = configureStore({
      preloadedState: initialState,
      reducer: { networkDrawer: networkDrawer.reducer }
    })
    const component = (
      <Provider store={store}>
        <NetworkDrawer />
      </Provider>
    );
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('should display txid and click', async () => {
    const initialState: Partial<RootState> = {
      networkDrawer: {
        height: 49,
        isOpen: true,
        title: 'Transaction completed',
        isLoading: false,
        txid: 'aoiwidwoajdpaowdpo123123apdowipo123'
      }
    };
    const store = configureStore({
      preloadedState: initialState,
      reducer: { networkDrawer: networkDrawer.reducer }
    })
    const component = (
      <Provider store={store}>
        <NetworkDrawer />
      </Provider>
    );
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
    const receiveButton = await rendered.findByTestId('networkDrawer_explorer')
    fireEvent.press(receiveButton)
    expect(Linking.canOpenURL).toBeCalled()
  })

  it('should display completed state and close panel', async () => {
    const initialState: Partial<RootState> = {
      networkDrawer: {
        height: 49,
        isOpen: true,
        title: 'Transaction completed',
        isLoading: false,
        txid: 'aoiwidwoajdpaowdpo123123apdowipo123'
      }
    };
    const store = configureStore({
      preloadedState: initialState,
      reducer: { networkDrawer: networkDrawer.reducer }
    })
    const component = (
      <Provider store={store}>
        <NetworkDrawer />
      </Provider>
    );
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
    const receiveButton = await rendered.findByTestId('networkDrawer_close')
    fireEvent.press(receiveButton)
    expect(Linking.canOpenURL).toBeCalled()
  })
})
