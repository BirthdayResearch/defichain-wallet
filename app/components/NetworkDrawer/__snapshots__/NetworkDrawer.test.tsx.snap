// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`networkDrawer should display completed state and close panel 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "borderBottomColor": "rgba(229, 231, 235, 1)",
      "borderLeftColor": "rgba(229, 231, 235, 1)",
      "borderRightColor": "rgba(229, 231, 235, 1)",
      "borderTopColor": "rgba(229, 231, 235, 1)",
      "borderTopWidth": 1,
      "bottom": 0,
      "flexDirection": "row",
      "height": 75,
      "paddingBottom": 12,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 12,
      "position": "absolute",
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#02B31B",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "material",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginRight": 4,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "600",
          },
          Object {
            "fontSize": 14,
            "fontWeight": "700",
            "lineHeight": 20,
          },
        ]
      }
    >
      Transaction completed
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
      testID="networkDrawer_explorer"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Array [
              Object {
                "color": "#ff00af",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 20,
                "marginRight": 4,
              },
            ],
          ]
        }
      >
        aoiwidwoajdpaow...
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ff00af",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(209, 213, 219, 1)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(209, 213, 219, 1)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(209, 213, 219, 1)",
        "borderRightWidth": 1,
        "borderTopColor": "rgba(209, 213, 219, 1)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      }
    }
    testID="networkDrawer_close"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "600",
          },
          Array [
            Object {
              "color": "#ff00af",
            },
            Object {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ],
        ]
      }
    >
      OK
    </Text>
  </View>
</View>
`;

exports[`networkDrawer should display loading state 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "borderBottomColor": "rgba(229, 231, 235, 1)",
      "borderLeftColor": "rgba(229, 231, 235, 1)",
      "borderRightColor": "rgba(229, 231, 235, 1)",
      "borderTopColor": "rgba(229, 231, 235, 1)",
      "borderTopWidth": 1,
      "bottom": 0,
      "flexDirection": "row",
      "height": 75,
      "paddingBottom": 12,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 12,
      "position": "absolute",
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginRight": 4,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "600",
          },
          Object {
            "fontSize": 14,
            "fontWeight": "700",
            "lineHeight": 20,
          },
        ]
      }
    >
      Loading...
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
      testID="networkDrawer_explorer"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Array [
              Object {
                "color": "#ff00af",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 20,
                "marginRight": 4,
              },
            ],
          ]
        }
      >
        ...
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ff00af",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`networkDrawer should display txid and click 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "borderBottomColor": "rgba(229, 231, 235, 1)",
      "borderLeftColor": "rgba(229, 231, 235, 1)",
      "borderRightColor": "rgba(229, 231, 235, 1)",
      "borderTopColor": "rgba(229, 231, 235, 1)",
      "borderTopWidth": 1,
      "bottom": 0,
      "flexDirection": "row",
      "height": 75,
      "paddingBottom": 12,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 12,
      "position": "absolute",
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#02B31B",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "material",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginRight": 4,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "600",
          },
          Object {
            "fontSize": 14,
            "fontWeight": "700",
            "lineHeight": 20,
          },
        ]
      }
    >
      Transaction completed
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
      testID="networkDrawer_explorer"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Array [
              Object {
                "color": "#ff00af",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 20,
                "marginRight": 4,
              },
            ],
          ]
        }
      >
        aoiwidwoajdpaow...
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ff00af",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(209, 213, 219, 1)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(209, 213, 219, 1)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(209, 213, 219, 1)",
        "borderRightWidth": 1,
        "borderTopColor": "rgba(209, 213, 219, 1)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      }
    }
    testID="networkDrawer_close"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "600",
          },
          Array [
            Object {
              "color": "#ff00af",
            },
            Object {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ],
        ]
      }
    >
      OK
    </Text>
  </View>
</View>
`;

exports[`networkDrawer should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "borderBottomColor": "rgba(229, 231, 235, 1)",
      "borderLeftColor": "rgba(229, 231, 235, 1)",
      "borderRightColor": "rgba(229, 231, 235, 1)",
      "borderTopColor": "rgba(229, 231, 235, 1)",
      "borderTopWidth": 1,
      "bottom": 0,
      "flexDirection": "row",
      "height": 75,
      "paddingBottom": 12,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 12,
      "position": "absolute",
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <Text />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginRight": 4,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "600",
          },
          Object {
            "fontSize": 14,
            "fontWeight": "700",
            "lineHeight": 20,
          },
        ]
      }
    >
      Loading...
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
      testID="networkDrawer_explorer"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            },
            Array [
              Object {
                "color": "#ff00af",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 20,
                "marginRight": 4,
              },
            ],
          ]
        }
      >
        ...
      </Text>
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(209, 213, 219, 1)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(209, 213, 219, 1)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(209, 213, 219, 1)",
        "borderRightWidth": 1,
        "borderTopColor": "rgba(209, 213, 219, 1)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      }
    }
    testID="networkDrawer_close"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "600",
          },
          Array [
            Object {
              "color": "#ff00af",
            },
            Object {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ],
        ]
      }
    >
      OK
    </Text>
  </View>
</View>
`;
